packages/@tinacms/cli/src/cmds/query-gen/genTypes.ts

```diff
@@ -48,8 +48,10 @@ export async function genClient(
     )
   }
 
+  const portIndex = process.argv.findIndex((a) => a === '--port') + 1
+  const PORT = (portIndex ? process.argv[portIndex] : null) || 4001
```

Comment:
The arguments are already being parsed with commander so we should not be doing our own parsing logic. 

It would be best to use the port option available at `options?.port`. That can [be accessed in this function](https://github.com/tinacms/tinacms/blob/feat/configurable-port-for-yarn-dev/packages/@tinacms/cli/src/buildTina/index.ts#L73) and will have to be passed to `genClient`

Reply:
@logan-anderson , I passed the port to `genClient`. Plz review again

Reply:
Found another issue.  See https://github.com/tinacms/tinacms/pull/3190#pullrequestreview-1109631857